### ISSUE TITLE
Allow users to display `alt` texts to extend support for UAAG Guideline 1.1: Alternative content

### ISSUE THREAD

**Issue by gautierchomel** [edrlab/thorium-reader#3139]
UAAG [1.1.1 Render Alternative Content](https://www.w3.org/TR/2015/NOTE-UAAG20-Reference-20151215/#sc_111) :
> The user can choose to render any type of [recognized](https://www.w3.org/TR/2015/NOTE-UAAG20-Reference-20151215/#def-recognize) [alternative content](https://www.w3.org/TR/2015/NOTE-UAAG20-Reference-20151215/#def-conditional-content) that is present for a content [element](https://www.w3.org/TR/2015/NOTE-UAAG20-Reference-20151215/#def-element). (Level A)

This can be considered as implemented by Thorium Reader, as users can access `alt` text through TTS. However, that support can be enhanced by providing the `alt` text in the image viewing modal, I suggest as a `detail` element.

**Comment by danielweck** [comment]
Related:

https://github.com/edrlab/r2-navigator-js/issues/76

https://github.com/edrlab/thorium-reader/issues/1717

https://github.com/edrlab/thorium-reader/issues/3139

https://github.com/edrlab/thorium-reader/issues/1983

https://github.com/edrlab/thorium-reader/issues/1974

Bonus headache :)

`<span role="img" aria-label="redacted text">&#160;&#160;&#160;&#160;</span>`

**Comment by gautierchomel** [comment]
Based on the first test implementation in #3357.

## Template proposal

```html
<figure>
<img href="..." alt "...">
<figcaption>{ content of figcaption }</figcaption>

<p id="alttext" aria-hidden="true">{ content of alt }</p> <!-- hidden from AT to avoid repetition -->
<p id="aria-labelledby">{ content of the element with id referenced in labelledby }</p> <!-- hidden from AT to avoid repetition -->
<p id="aria-describedby">{ content of the element with id referenced in describedby }</p>
<p id="aria-details">{ content of the element with id referenced in aria-details }</p>
</figure>
```

### Explanations

* There should be only one of `alt` or `labelledby` in publications.
* There should be only one of `describedby` or `details`  in publications.
* The good practice in production is to make all those contents explicits alone, so we shall not add verbosity.
* Using list would add semantics that does not match the intent, and most cases will have only one or two elements.


### Example

```html
<!-- Main content -->
<figure>
<img id="img1" src="figure1.png" alt="Schematic of the device" aria-details="extdesc-1">
<figcaption>figure 1</figcaption>
<a id="extdesc-1" href="extended-descriptions.xhtml#desc-img1">Extended description</a>
</figure>
```

Would render as

```html

<figure>
<img id="img1" src="figure1.png" alt="Schematic of the device" aria-details="extdesc-1">
<figcaption>figure 1</figcaption>
<p id="alttext" aria-hidden="true">Schematic of the device</p>
<p id="aria-details"><a id="extdesc-1" href="extended-descriptions.xhtml#desc-img1">Extended description</a></p>
</figure>

```

## Considerations

* We discussed using `details`/`summary` with the `alt` content as the `summary`, but this seems risky because:
    * it does not explicitly describe the action attached to activating the `summary`;
    * there is no explicit limit on `alt` length (we may end up with a large `summary`);
    * it would require `aria-hidden` on the image itself.

* In the near future, we hope to have two new ARIA roles to securely identify the extended description and display it there instead of the link.
    * This is described by the [Explainer: Extended Descriptions for Publications and the Web](https://github.com/w3c/publ-a11y-extended-desc?tab=readme-ov-file#proposed-approach). 
    * A [POC EPUB](https://github.com/daisy/transitiontoepub/tree/main/experiments/extended-desc) is available to test implementation.

**Comment by arthur-lemeur** [comment]
This proposal seems complicated to me regarding its implementation in the dialog, the image is in its own component which handle the zoom in/out and the informations displayed are in their own component too, elsewhere in the code. An html rendering like your proposition doesn't not seem possible in the current state.

**Comment by gautierchomel** [comment]
I see. Then, we should keep only the `p`' like : 

```html
<p id="figcaption">{ content of figcaption }</p>
<p id="alttext" aria-hidden="true">{ content of alt }</p> <!-- hidden from AT to avoid repetition -->
<p id="aria-labelledby">{ content of the element with id referenced in labelledby }</p> <!-- hidden from AT to avoid repetition -->
<p id="aria-describedby">{ content of the element with id referenced in describedby }</p>
<p id="aria-details">{ content of the element with id referenced in aria-details }</p>
```

**Comment by gautierchomel** [comment]
Files for testing:
* [AccessibleImageSampleBook_v2.2.epub](https://github.com/benetech/AccessibleImageSampleBook/blob/master/AccessibleImageSampleBook_v2.2.epub) - several images with `alt` and `captions`.
* State of the art epubtest [Extended Descriptions](https://www.epubtest.org/test-books/extended-descriptions)
* [Extended description POC EPUB](https://github.com/daisy/transitiontoepub/tree/main/experiments/extended-desc) - include `aria-details`

**Comment by gautierchomel** [comment]
After the draft implementation, several questions arrised:
* The panel does not allow to display authored CSS, meaning that any HTML present in `figcaption` or `aria-detail` will end up unstylled unless we provide an opinionated CSS. 
* Sanitisation on HTML is necessary to display anything in this panel.
* Displaying only the `alt` in the modal will let it uncontextualised.

For future implementation we should consider:
* Triggering display of the `alt`in the main reading view, with an affordance similar to the annotation display/mask.
* Extended Description as described in the W3C explainer cannot be implemented in a panel view as it will be expected to have it styled.
* Textual information attached to an image should be thought separately from the Zoom in Image Panel.
